### PR TITLE
Bump Node version to 16.15.0

### PR DESCRIPTION
**Pull request by nefeline**
Bump [Node version](https://nodejs.org/en/blog/release/v16.15.0) from `16.13.2` to `16.15.0`.

Context: `16.14.0` is the minimum version required for bumping `lint-staged` [to the latest version](https://github.com/woocommerce/woocommerce-blocks/pull/8793).

We also have [ongoing work](https://github.com/woocommerce/woocommerce-blocks/pull/8839) relying on `usehooks-ts@2.9.1`, which requires Node version `16.15.0`.

#### User Facing Testing

* Check that all the Github actions are green
* Pull this branch to your local machine
* `nvm install` or `nvm use` should switch you to use version ^16.15.0
* `npm ci` should run successfully
* `npm run build` should build successfully
* `npm run start` should run successfully
* `npm run test` should run successfully
*  Watch for any other additional warnings/errors in the console while running any builds

### Changelog

> Updated to node version ^16.15.0

### CODE REVIEW

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @nefeline! I've successfully tested the change. Let's wait for other's to reconfirm, but you have my approval already. 🙌

**Review by thealexandrelara — APPROVED** [review]
Thank you @nefeline for working on this!! Just tested and it is working great!

**Comment by opr** [comment]
Hey @nefeline I would skip this type of PR from the changelog. I will apply the label now.

**Pull request merged**
